### DISCUSSION TITLE
Struggling with testing a console app

### DISCUSSION THREAD

**Discussion by samsmithnz** [spectreconsole/spectre.console#912] [Q&A] [answered]
Hi, I'm struggling to get testing to work. I have a table my console app is generating, now I'd like to create a test to verify the output, but I can't find an example in the docs.

Currently my test looks a bit like: 
```
            //Act
            AnsiConsole.Record();
            Program.Main(parameters);

            //Asset
            var text = AnsiConsole.ExportText();
            Assert.IsNotNull(text);
```

The issue is that the text, for some reason, is `...`, but when I run the console app with the same settings, it's a full table. Is there something going on here I'm missing? Is there another way I should be writing this test?

**Comment by patriksvensson** [comment, marked as answer]
@samsmithnz i wouldn't recommend using the `AnsiConsole` for tests. 

Take a look at the tests in Spectre.Console. We use the `TestConsole` part of the `Spectre.Console.Testing` package.

**Reply by samsmithnz** [reply]
Are there docs for tests I'm missing? I can find very little in these docs and in google searches adding to my confusion. The actual unit tests in this project don't help either.

**Reply by samsmithnz** [reply]
To be clear, I think there are two questions here
1. the table is returning `...` in a test, whereas it returns a full table when I run it interactively. Why is this? What is it I don't understand about AnsiConsole
2. is the documentation for testing, which seems to be very very light, or redirections to look at the source code of this module - which doesn't answer my questions...

**Comment by FrankRay78** [comment]
Hello @samsmithnz, I plan to address this discussion more thoroughly in PR https://github.com/spectreconsole/spectre.console/pull/1631 over the coming weeks.
